### PR TITLE
Add optional err-id to response

### PR DESCRIPTION
**Pull request by fw42**
@wisq @arthurnn

Add err-id from API response to return value. Make it optional to not break backwards compatibility.

Stupid GitHub UI, accidentally opened the PR to upstream :-)

### CODE REVIEW

**Comment by arthurnn** [comment]
LGTM!

**Comment by fw42** [comment]
@camilo, quick look please?

File: lib/airbrake/sender.rb [hunk 1]
```diff
@@ -50,8 +50,14 @@ def send_to_airbrake(data)
       end
 
       if response && response.respond_to?(:body)
-        error_id = response.body.match(%r{<_id[^>]*>(.*?)</_id>})
-        error_id[1] if error_id
+        notice_id = response.body.match(%r{<_id[^>]*>(.*?)</_id>})
```
Comment (sirupsen) [inline lib/airbrake/sender.rb:53]: Is the response body JSON? This is super ugly..

File: lib/airbrake/sender.rb [hunk 2]
```diff
@@ -50,8 +50,14 @@ def send_to_airbrake(data)
...
+        error_id = response.body.match(%r{<err-id[^>]*>(.*?)</err-id>})
+
+        if with_error_id
+          return notice_id.try(:[], 1), error_id.try(:[], 1)
+        else
+          return notice_id.try(:[], 1)
+        end
```
Comment (sirupsen) [inline lib/airbrake/sender.rb:60]: You don't need the returns

File: lib/airbrake/sender.rb (hunk 1 above)
Comment (fw42) [inline lib/airbrake/sender.rb:53]: No, it's XML. And yes, it's ugly (but cheap).

Comment (sirupsen) [inline lib/airbrake/sender.rb:53]: Ok.. go for it

**Comment by sirupsen** [comment]
Fix the return thing, then ship

File: lib/airbrake/sender.rb (hunk 1 above)
Comment (camilo) [inline lib/airbrake/sender.rb:53]: lol @csfrancis would be proud

**Pull request merged**